### ISSUE TITLE
RouteEvent from .AnyState

### ISSUE THREAD

**Issue by AlDrago** [ReactKit/SwiftState#20]
Hi, 
is it a bug the that last line of the code below returns `false` or I misunderstood something?

```Swift
enum State : StateType {
    case A, B, C
    case AnyState
    
    init(nilLiteral: Void) {
        self = AnyState
    }
}

enum Event: StateEventType {
    case Event1
    case AnyEvent
    
    init(nilLiteral: Void) {
        self = AnyEvent
    }
}

let machine = StateMachine<State, Event>(state: State.A) { machine in
    machine.addRoute(.A => .B)
    machine.addRoute(.AnyState => .C)
    machine.addRouteEvent(.Event1, transitions: [.AnyState => .A])
}

machine <- .C		// Returns true
machine <-! .Event1	// Returns false ... ?bug?
```

**Comment by inamiy** [comment]
Thanks for report.
You've done correctly, and it is a bug, so I will investigate more.

**Comment by inamiy** [comment]
Fixed in 90fab1f :+1:

**Comment by AlDrago** [comment]
great, thank you!
